app/models/invenio_rdm_record_converter.rb

```diff
@@ -54,7 +54,7 @@ def generic_file_content_path(checksum)
 
   def record_for_export(generic_file)
     {
-      "pids": invennio_pids(generic_file.doi.shift),
+      "pids": invenio_pids(generic_file.doi.shift),
```

Comment:
Did we ever figure out what was going on with shift here? I looked at this a little bit thinking that maybe the generic_file argument was a copy of the generic_file, not a reference, and that would mean the mutation was only happening to a copy. I didn't actually confirm this though.

Reply:
Yes, I looked into this. It's actually because `generic_file` here is an ActiveFedora object. Calling `shift` won't mutate because it does not trigger it's datastore update.

Reply:
Curious about that shift not updating the datastore: 
- would the changes still be in memory though?
- would make sense to me that they are not if: 
  * `generic_file.doi` is a join everytime (although I doubt it) OR
  * `generic_file` here is not the same in-memory `gf` as below. In other words there is 2 DB fetches of the same file OR
  * that shift has been overridden

Just curious :thinking: .

Reply:
It is very misleading/confusing. I am glad you caught that `shift` should mutate under normal circumstances. Otherwise I may have continued to use in other cases with unintended consequences. 

This is what I know about Active**Fedora** and Active**Record**: is that they don't have the same interface, with ActiveFedora being much more limited. I have had trouble in the past with things like using `<<` to append to what should be an Array, but when trying to commit/save, it didn't. So, my guess is that `shift` returned the value first, then tried to remove the item from the "Array"(the commit/save part), but exhibited the same behavior as when I tried using the shovel operator. That's what I meant by `shift` not triggering the update, but should have said failed to make the update. So, it must be that `shift` is overriden.

Reply:
Really cool to know. Thanks for the explanation @Meowcenary.